bolero/representation/promp_behavior.py

```diff
@@ -0,0 +1,497 @@
+        we take the first position as x0.
+
+        Parameters
+        ----------
+        inputs : array-like, shape = (2 * n_task_dims,)
+            Contains positions and velocities in that order.
+            Each type is stored contiguously, i.e. for n_task_dims=2 the order
+            would be: xxvv (x: position, v: velocity).
+        """
+        # just open loop by now
+
+    def get_outputs(self, outputs):
+        """Get outputs of the last step.
+
+        Parameters
+        ----------
+        outputs : array-like, shape = (2 * n_task_dims,)
+            Contains positions, velocities in that order.
+            Each type is stored contiguously, i.e. for n_task_dims=2 the order
+            would be: xxvv (x: position, v: velocity).
+        """
+        i = int(self.t/self.dt)
+        outputs[:self.n_task_dims] = self.y[i]#self.valueMeans[::2]#self.y[:]
+        outputs[self.n_task_dims:2*self.n_task_dims] = self.yd[i]#self.valueMeans[1::2]#self.yd[:]
+        if self.useCovar:
+            outputs[2*self.n_task_dims:] = self.covars[i].flatten()
+
+    def step(self):
+        """Compute desired position, velocity and acceleration."""
+        if self.n_task_dims == 0:
+            return
+
+        #self.data.step(self.t,self.valueMeans)#,self.valueCovs)
+
+        if self.t == self.last_t:
+            self.last_t = -1.0
+        else:
+            self.last_t = self.t
+            self.t += self.dt
+
+    def can_step(self):
+        """Returns if step() can be called again.
+
+        Note that calling step() after this function returns False will not
+        result in an error. The velocity and acceleration will be set to 0
+        and we hold the last position instead.
+
+        Returns
+        -------
+        can_step : bool
+            Can we call step() again?
+        """
+        return self.t <= self.execution_time
+
+    def get_n_params(self):
+        """Get number of weights.
+
+        Returns
+        -------
+        n_params : int
+            Number of promp weights
+        """
+        random_variables = len(self.data.mean_)
+        if self.learnCovariance:            
+            correlation_coefficients = (len(self.data.covariance_)-random_variables)/2
+            return 2*random_variables + correlation_coefficients
+        else:
+            return random_variables
+
+    def get_params(self):
+        """Get current weights. 
+
+        Returns
+        -------
+        params : array-like, shape = (n_params,)
+            Current weights
+        """
+        random_variables = self.data.mean_
+        if self.learnCovariance:
+            cov = np.array(self.data.covariance_).reshape(len(random_variables),len(random_variables))
+            Dinv =np.linalg.inv(np.sqrt(np.diag(np.diag(cov))))
+            cor = Dinv.dot(cov.dot(Dinv))
+
+            return random_variables + np.sqrt(np.diag(cov)).tolist() + np.arctanh(cor[np.tril_indices(len(random_variables),k=-1)]).tolist()
```

Comment:
Hardly readable and line too long.

Reply:
also: why does it have to be a list? you can concatenate numpy arrays